### DISCUSSION TITLE
Blocked my slingboxes from internet.  Here is what I found.

### DISCUSSION THREAD

**Discussion by mdehrlich** [GerryDazoo/Slinger#289] [General]
I set my firewall to block all inbound and outbound packets to/from my slingboxes.  My security log shows all blocked traffic.  I show no inbound messages to the boxes, however, many outbound messages to various IP addresses and ports from the slingboxes.  I wonder what they will do when no one answers their calls to home?  Right now, everything is working ok with slinger.  I am seeing about 10 outbound messages per box per minute.

**Comment by sp222222** [comment]
KEEP blocking those outbounds... that is what bricks the boxes.

**Comment by robosmith** [comment]
What kind of Slingboxes?

**Reply by mdehrlich** [reply]
a 350 and a M1

**Reply by GerryDazoo** [reply]
I wouldn't worry about a self destruct. Otherwise boxes would have been going done anytime they lost their internet connect for what ever reason.

**Reply by surveyor2** [reply]
Before I caught up on recent threads I perhaps foolishly connected to sling last night to see if it had been disconnected yet and it was working as normal. Fortunately no harm done to my M1's because they're all still working okay with slinger. The firewall on my ArcherVR900 (now TPlink) router is hopeless with devices either blocked or not blocked. No incoming or outgoing configuration. I've switched off uPNP but I don't know if that is sufficient to prevent the M1's phoning home. Do I need to get some extra kit? There is just one port being forwarded to the Slinger server and everything seems to be okay with connections from inside and outside the LAN.

James

**Reply by GerryDazoo** [reply]
Goggle up the manual and checkout parental controls. I think that's what you need to setup.

**Reply by surveyor2** [reply]
Thanks.

**Reply by purdyson4** [reply]
> Goggle up the manual and checkout parental controls. I think that's what you need to setup.

Hi Gerry, could you assist me in configuring my sb 500
John

**Reply by womeara6** [reply]
Gerry, I just read the Lightreading article on Dish bricking the Solo and Pro's.  You mention fireewall protections especially to prevent Dish from attacking the 500s.  Can you provide the specifics?  I've been learning alot in this process but not enough to be safe.

Thanks in advance.
Bill

**Reply by sp222222** [reply]
> Gerry, I just read the Lightreading article on Dish bricking the Solo and Pro's. You mention fireewall protections especially to prevent Dish from attacking the 500s. Can you provide the specifics? I've been learning alot in this process but not enough to be safe.
> 
> Thanks in advance. Bill

Bill:

Same song and dance if you read around. take the router you have the slingbox directly connected to and make a rule to block port 5678 on the IP that the slingbox itself uses. It's that simple.

**Reply by sp222222** [reply]
but block outgoing and incoming. I got roasted because I only had incoming blocked.

**Comment by womeara6** [comment]
Hi Gerry. Thanks for your input.  This is where I  am having difficulty.  I
have a Linksys WRT1900AC router. Supposed to be good.  Unfortunately, I
can’t find in the document how to block the port on the IP of my Slingbox.
I found a document on Github for your Slinger that goes through all the
steps to set up the Slinger Server and discusses blocking internet access
for the slingbox device on my network and 2 specific websites, slingbox.com
and sling.com through the use of the parental controls.  That I have done.
But not sure if it works.  I would also like to block the specific port that
is mentioned.

 

Under Security tab I have the ability to perform port forwarding.  This
allows for specific rules on ports and ip addresses.  I could add the port
5678 here and not enable it but the documents says that  If you don’t want
to use port forwarding but want to keep the information in the table,
unselect the checkbox.  Under enabled, it reports False.  So I don’t think
it blocks the port.  

 

Any help with blocking would be appreciated or if you think I am covered
with the parental controls.  

 

By the way, I’m not sure when the Solo’s got blocked but mine is alive but
dead.  It shows up on the network with an IP.  I had gotten the password and
finderid using the Slingplayer desktop method.  I’ve been working with the
500 since last week but today I tried logging into the Solo, Password
reported as incorrect.  So it looks like I got the brick program.

 

Thanks in advance,

Bill

**Reply by sp222222** [reply]
it got bricked the moment you put your solo onto the netwotk without proper blocking rules, after Nov9th. welcome to the brick solo club. the solo will ping but if port 5678 was not blocked on your router both inbound and outbound, the solo is a doorstop for now. Port forward does nothing to block port. nothing.

**Reply by Toei79** [reply]
> it got bricked the moment you put your solo onto the netwotk without proper blocking rules, after Nov9th. welcome to the brick solo club. the solo will ping but if port 5678 was not blocked on your router both inbound and outbound, the solo is a doorstop for now. Port forward does nothing to block port. nothing.

what boxes so far are sensible to get bricked?

im soon to connect one sb300 and one 500 , i suppose i need block port 5678 both ways for each.

**Comment by Pthepeg** [comment]
Here is what I found for my setup:
Parental controls on my Linksys router only block outgoing web browsing (port 80) and not the other ports.
Using pi-hole I discovered for several weeks, my slingbox350 was generating plenty of traffic to several sling.com subdomains but fortunately is still not bricked. 
Who knows if the 350 will be permanently immune to bricking but I've now blacklisted sling.com in pi-hole, just in case.

**Reply by purdyson4** [reply]
Hi all, (John here, sb500 MacBook Air), I have terminal/homebrew all in ready to continue, any further help would be appreciated

**Reply by DoogieHowitzerWMD** [reply]
@purdyson4 That's off-topic for this thread. Head back to #208 where theweatherguy was helping walk you through things.

**Comment by womeara6** [comment]
Hi Gerry.  Time with Linksys was not very informative.  Seems that their WRT1900AC can’t block ports.  I’m not sure it can handle Parental Controls either.  According to the Linksys rep…

 

“I do apologize, but based on our system. You cannot block a port number in the Linksys router, this router is only capable of opening the port numbers that were being blocked by your Internet Provider.

 

Some network and security administrators view port blocking as a critical tool for securing systems and information, and see it as part of the Internet Service Provider's mission to manage the security risk to its users from theft and destruction of personal information, business records, and other critical electronic forms of information.”

 

Not sure I agree with Linksys.  I probably need to look for a new router that can provide security for my network.

 

Bill

 

From: ***@***.*** ***@***.***> 
Sent: Sunday, December 4, 2022 8:40 PM
To: 'GerryDazoo/Slinger' ***@***.***>; 'GerryDazoo/Slinger' ***@***.***>
Cc: 'Comment' ***@***.***>
Subject: RE: [GerryDazoo/Slinger] Blocked my slingboxes from internet. Here is what I found. (Discussion #289)

 

Hi Gerry. Thanks for your input.  This is where I  am having difficulty.  I have a Linksys WRT1900AC router. Supposed to be good.  Unfortunately, I can’t find in the document how to block the port on the IP of my Slingbox.  I found a document on Github for your Slinger that goes through all the steps to set up the Slinger Server and discusses blocking internet access for the slingbox device on my network and 2 specific websites, slingbox.com and sling.com through the use of the parental controls.  That I have done.  But not sure if it works.  I would also like to block the specific port that is mentioned.

 

Under Security tab I have the ability to perform port forwarding.  This allows for specific rules on ports and ip addresses.  I could add the port 5678 here and not enable it but the documents says that  If you don’t want to use port forwarding but want to keep the information in the table, unselect the checkbox.  Under enabled, it reports False.  So I don’t think it blocks the port.  

 

Any help with blocking would be appreciated or if you think I am covered with the parental controls.  

 

By the way, I’m not sure when the Solo’s got blocked but mine is alive but dead.  It shows up on the network with an IP.  I had gotten the password and finderid using the Slingplayer desktop method.  I’ve been working with the 500 since last week but today I tried logging into the Solo, Password reported as incorrect.  So it looks like I got the brick program.

 

Thanks in advance,

Bill

**Reply by GerryDazoo** [reply]
“I do apologize, but based on our system. You cannot block a port number in the Linksys router, this router is only capable of opening the port numbers that were being blocked by your Internet Provider.

I have no idea what this means. And the guy who wrote it probably doesn't either.

**Comment by theweatherguy** [comment]
Hi Gerry,
I wanted to share a discovery with you and the group that I made over this weekend.  I don’t know the significance of it, and I haven’t seen anyone mention this to date.  I’m wondering if it’s a clue to how some slingboxes are being bricked. 

I have an old Mac that is running El Capitan from over 7 years ago. (I keep the Mac to run very old software.  It’s on it's own,  _isolated_ network but connected to the internet.)

After I mailed you my unbricked Solo box (hopefully you’ll be receiving that soon) I decided to boot up this Mac to see what I could glean from the slingplayer app console logs.   I decided to launch the app just to see what messages were in the logs. Not being connected to any slingbox since 2015, I didn’t see much. 

BUT I noticed something very curious…. The slingplayer app had a modified date for the current date

I don’t know about Windows, but regular Mac apps don’t changed their “modified dates” when they've been launched. **This means that the application went out to the Internet and somehow updated and modified itself in the background**. 

I'm wondering if some boxes are being bricked, not from their direct connection to internet, but rather from a slingplayer app being modified. 

Maybe people should remove their slingplayer apps?

Just something to consider.

**Reply by GerryDazoo** [reply]
Yes. This is why a month ago I recommended NOT running any sling supplied software, It is a possible attack vector if sling wanted to use it. But in the interim we have no choice because we don't have all the code we need to do it ourselves.
I'm of the opinion that Mr. Sling achieved his goals when he bricked the solos, pro and proHDs. So they couldn't stream and disabled their support for streaming on the newer boxes. The fact that nothing bad has happened since N9 is encouraging.  Also there would be significant cost to them to make and release new versions of the desktop code just to shutdown our merry little band of enthusiasts.

**Reply by DoogieHowitzerWMD** [reply]
> I'm of the opinion that Mr. Sling achieved his goals when he bricked the solos, pro and proHDs. So they couldn't stream and disabled their support for streaming on the newer boxes. The fact that nothing bad has happened since N9 is encouraging. Also there would be significant cost to them to make and release new versions of the desktop code just to shutdown our merry little band of enthusiasts.

You, and probably most of the smarter contributors here, realized this weeks ago, but I think it makes sense to me now why Mr Sling aimed to brick those models in particular. The newer models all had to pass their feed through sling's servers, so shutting the servers down disabled those boxes' functionality (without third-party software like Slinger, that is).

However, with the right setup, those older boxes (could?) still have a way to retain all of the same functionality **without** third-party software. From my understanding (and correct me if I'm wrong), it should be feasible to put a properly configured ProHD behind a VPN server so that a client computer with Slingplayer 2.0.4 could view the stream from pretty much anywhere. Mr Sling worrying about that would seem to jive with why they bricked the models with that capability, and maybe why we haven't seen anything else nefarious from them since N9.

**Reply by GerryDazoo** [reply]
You mean no third-party software besides a VPN server and client. :) And it won't work with newer boxes that need SPD 5.0.x
You still have to make sure the slingbox can't send messages to Mr. Sling.

**Reply by theweatherguy** [reply]
As a followup to my observation, I'm wondering if there's any value in comparing the "modified" slingplayer app with an unmodified copy of the same app from a backup I have?   Specifically,  I'm wondering if any very computer-advanced people on this forum are able to use a hex-editor, to compare the modified slingplayer app with the backup copy (unmodified).    

Would that provide any specific insight into what changes Mr. Sling is sending out?   Don't know if that's even feasible, but I'd be happy to post my new modified copy with along with the unmodified from a backup if anyone is able to glean and decipher changes that would be meaningful.  

Just trying to help...

**Comment by womeara6** [comment]
Hi Gerry. Back again.  Just spent some time with a Linksys technician on this port blocking issue.  Here is what he provided.

 

Most Linksys routers/firewalls rely on simple NAT routing and basic port filtering to control the flow of traffic through the router. Depending on the direction of the traffic flow, a different filtering methodology is applied.

Filtering Traffic from External Sources

Linksys adheres to the minimalist approach to filtering when it comes to filtering traffic from external sources. By default, all traffic that originates from an external host is blocked by the router/firewall unless it is specifically permitted. This policy ensures that only the traffic you explicitly permit is allowed to access protected resources. Linksys provides three methods of explicitly permitting traffic:

*        Port-range forwarding

*        Port triggering

*        DMZ forwarding

I have port forwarding set up for the 5201 on my 500 and had 5001 for my Solo.  Both worked fine pre 9N.

 

So, it appears to me that the Linksys router should be blocking port 5678 by default with no additional router configuration needed.

 

This however doesn’t explain the Solo getting bricked except that the Solo was attached via an ethernet cable.

 

Can you give me your thoughts?

 

Thanks in advance!

Bill

 

 

From: ***@***.*** ***@***.***> 
Sent: Monday, December 5, 2022 3:15 PM
To: 'GerryDazoo/Slinger' ***@***.***>; 'GerryDazoo/Slinger' ***@***.***>
Cc: 'Comment' ***@***.***>
Subject: RE: [GerryDazoo/Slinger] Blocked my slingboxes from internet. Here is what I found. (Discussion #289)

 

Hi Gerry.  Time with Linksys was not very informative.  Seems that their WRT1900AC can’t block ports.  I’m not sure it can handle Parental Controls either.  According to the Linksys rep…

 

“I do apologize, but based on our system. You cannot block a port number in the Linksys router, this router is only capable of opening the port numbers that were being blocked by your Internet Provider.

 

Some network and security administrators view port blocking as a critical tool for securing systems and information, and see it as part of the Internet Service Provider's mission to manage the security risk to its users from theft and destruction of personal information, business records, and other critical electronic forms of information.”

 

Not sure I agree with Linksys.  I probably need to look for a new router that can provide security for my network.

 

Bill

 

From: ***@***.*** ***@***.***>  ***@***.*** ***@***.***> > 
Sent: Sunday, December 4, 2022 8:40 PM
To: 'GerryDazoo/Slinger' ***@***.*** ***@***.***> >; 'GerryDazoo/Slinger' ***@***.*** ***@***.***> >
Cc: 'Comment' ***@***.*** ***@***.***> >
Subject: RE: [GerryDazoo/Slinger] Blocked my slingboxes from internet. Here is what I found. (Discussion #289)

 

Hi Gerry. Thanks for your input.  This is where I  am having difficulty.  I have a Linksys WRT1900AC router. Supposed to be good.  Unfortunately, I can’t find in the document how to block the port on the IP of my Slingbox.  I found a document on Github for your Slinger that goes through all the steps to set up the Slinger Server and discusses blocking internet access for the slingbox device on my network and 2 specific websites, slingbox.com and sling.com through the use of the parental controls.  That I have done.  But not sure if it works.  I would also like to block the specific port that is mentioned.

 

Under Security tab I have the ability to perform port forwarding.  This allows for specific rules on ports and ip addresses.  I could add the port 5678 here and not enable it but the documents says that  If you don’t want to use port forwarding but want to keep the information in the table, unselect the checkbox.  Under enabled, it reports False.  So I don’t think it blocks the port.  

 

Any help with blocking would be appreciated or if you think I am covered with the parental controls.  

 

By the way, I’m not sure when the Solo’s got blocked but mine is alive but dead.  It shows up on the network with an IP.  I had gotten the password and finderid using the Slingplayer desktop method.  I’ve been working with the 500 since last week but today I tried logging into the Solo, Password reported as incorrect.  So it looks like I got the brick program.

 

Thanks in advance,

Bill

**Reply by Toei79** [reply]
My humble thought  here its asus  make best routers, lot of features

**Comment by Pthepeg** [comment]
I'm going to chime in on this.
What you're describing is traffic ORIGINATING from external sources.
When traffic originates from a source behind the router, destined for an
external host, the router will by default, allow the external host response
to make it through to the internal host.

Blocking port 5678 is only going to block an external host ORIGINATING
traffic to an internal host. It won't stop slingbox calling out on port
5678, and wont stop the reply to that call (from sling.com) from reaching
the internal slingbox.

**Reply by Toei79** [reply]
> I'm going to chime in on this. What you're describing is traffic ORIGINATING from external sources. When traffic originates from a source behind the router, destined for an external host, the router will by default, allow the external host response to make it through to the internal host. Blocking port 5678 is only going to block an external host ORIGINATING traffic to an internal host. It won't stop slingbox calling out on port 5678, and wont stop the reply to that call (from sling.com) from reaching the internal slingbox.
> […](#)
> On Tue., Dec. 6, 2022, 12:31 p.m. womeara6, ***@***.***> wrote: Hi Gerry. Back again. Just spent some time with a Linksys technician on this port blocking issue. Here is what he provided. Most Linksys routers/firewalls rely on simple NAT routing and basic port filtering to control the flow of traffic through the router. Depending on the direction of the traffic flow, a different filtering methodology is applied. Filtering Traffic from External Sources Linksys adheres to the minimalist approach to filtering when it comes to filtering traffic from external sources. By default, all traffic that originates from an external host is blocked by the router/firewall unless it is specifically permitted. This policy ensures that only the traffic you explicitly permit is allowed to access protected resources. Linksys provides three methods of explicitly permitting traffic: * Port-range forwarding * Port triggering * DMZ forwarding I have port forwarding set up for the 5201 on my 500 and had 5001 for my Solo. Both worked fine pre 9N. So, it appears to me that the Linksys router should be blocking port 5678 by default with no additional router configuration needed. This however doesn’t explain the Solo getting bricked except that the Solo was attached via an ethernet cable. Can you give me your thoughts? Thanks in advance! Bill From: ***@***.*** ***@***.***> Sent: Monday, December 5, 2022 3:15 PM To: 'GerryDazoo/Slinger' ***@***.***>; 'GerryDazoo/Slinger' ***@***.***> Cc: 'Comment' ***@***.***> Subject: RE: [GerryDazoo/Slinger] Blocked my slingboxes from internet. Here is what I found. (Discussion #289) Hi Gerry. Time with Linksys was not very informative. Seems that their WRT1900AC can’t block ports. I’m not sure it can handle Parental Controls either. According to the Linksys rep… “I do apologize, but based on our system. You cannot block a port number in the Linksys router, this router is only capable of opening the port numbers that were being blocked by your Internet Provider. Some network and security administrators view port blocking as a critical tool for securing systems and information, and see it as part of the Internet Service Provider's mission to manage the security risk to its users from theft and destruction of personal information, business records, and other critical electronic forms of information.” Not sure I agree with Linksys. I probably need to look for a new router that can provide security for my network. Bill From: ***@***.*** ***@***.***> ***@***.*** ***@***.***> > Sent: Sunday, December 4, 2022 8:40 PM To: 'GerryDazoo/Slinger' ***@***.*** ***@***.***> >; 'GerryDazoo/Slinger' ***@***.*** ***@***.***> > Cc: 'Comment' ***@***.*** ***@***.***> > Subject: RE: [GerryDazoo/Slinger] Blocked my slingboxes from internet. Here is what I found. (Discussion #289) Hi Gerry. Thanks for your input. This is where I am having difficulty. I have a Linksys WRT1900AC router. Supposed to be good. Unfortunately, I can’t find in the document how to block the port on the IP of my Slingbox. I found a document on Github for your Slinger that goes through all the steps to set up the Slinger Server and discusses blocking internet access for the slingbox device on my network and 2 specific websites, slingbox.com and sling.com through the use of the parental controls. That I have done. But not sure if it works. I would also like to block the specific port that is mentioned. Under Security tab I have the ability to perform port forwarding. This allows for specific rules on ports and ip addresses. I could add the port 5678 here and not enable it but the documents says that If you don’t want to use port forwarding but want to keep the information in the table, unselect the checkbox. Under enabled, it reports False. So I don’t think it blocks the port. Any help with blocking would be appreciated or if you think I am covered with the parental controls. By the way, I’m not sure when the Solo’s got blocked but mine is alive but dead. It shows up on the network with an IP. I had gotten the password and finderid using the Slingplayer desktop method. I’ve been working with the 500 since last week but today I tried logging into the Solo, Password reported as incorrect. So it looks like I got the brick program. Thanks in advance, Bill From: sp222222 ***@***.*** ***@***.***> > Sent: Sunday, December 4, 2022 5:38 PM To: GerryDazoo/Slinger ***@***.*** ***@***.***> > Cc: womeara6 ***@***.*** ***@***.***> >; Comment ***@***.*** ***@***.***> > Subject: Re: [GerryDazoo/Slinger] Blocked my slingboxes from internet. Here is what I found. (Discussion #289) but block outgoing and incoming — Reply to this email directly, < [#289 (reply in thread)](https://github.com/GerryDazoo/Slinger/discussions/289#discussioncomment-4308498)> view it on GitHub, or < https://github.com/notifications/unsubscribe-auth/A4N5MTIM3554SHKWVMNAUWDWLUMN7ANCNFSM6AAAAAAR7JOSWY> unsubscribe. You are receiving this because you commented. < https://github.com/notifications/beacon/A4N5MTNUCXYHCJ7ZEYQEXPTWLUMN7A5CNFSM6AAAAAAR7JOSW2WGG33NNVSW45C7OR4XAZNRIRUXGY3VONZWS33OINXW23LFNZ2KUY3PNVWWK3TUL5UWJTQAIG7BE.gif> Message ID: < ***@***.***> ***@***.***> — Reply to this email directly, view it on GitHub <[#289 (comment)](https://github.com/GerryDazoo/Slinger/discussions/289#discussioncomment-4325952)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/A3TIBDLRKY7BRIP52ZQK2Q3WL6BBLANCNFSM6AAAAAAR7JOSWY> . You are receiving this because you commented.Message ID: ***@***.***>

You mean its useless block the port?

**Reply by DoogieHowitzerWMD** [reply]
> You mean its useless block the port? @Toei79 

No, it's semantics.

So, @womeara6's router is already blocking outside attempts from sling (or presumably, from anyone) from trying to initiate contact with the slingbox on their network. This is **INCOMING** connections that are being blocked. Most routers are doing this by default.

What is happening with the Slingboxes that are getting bricked is that **THEY ARE INITIATING THE CONNECTION**. The slingbox is reaching out to Sling ("phoning home" as Gerry says). Sling can then respond to that slingbox, and that response is permitted through the firewall.

The slingbox needs to be blocked from making **OUTGOING** connections (specifically, through external port 5678 on the router). Many routers, by default, are permissive of devices on your network initiating outgoing connections, for ease of usage and less configuration on the part of the end user.

**Comment by Pthepeg** [comment]
You would need to block outgoing traffic at that port to make it work.

I tried using Parental Controls on my Linksys. It does work, blocks
outgoing traffic but only on port 80, web browsing. I discovered my
slingbox was still calling out through the router, using non 80 ports.

**Comment by womeara6** [comment]
Doogie, so I can understand the ip traffic, the Solo initiated the connection to Sling and it was made from port 5678 and because Solo initiated the contact to Sling it opened the door for a reply from Sling using the same 5678 port. Sling then sent the killer to Solo.

 

Thanks,

Bill

**Reply by DoogieHowitzerWMD** [reply]
@womeara6 Might not necessarily be the same port on the reply (actually, I have no idea on that), but yes, the solo reached out to sling through external port 5678 on the router.
